web/src/engine/websites/MangaUpJapan.ts

```diff
@@ -0,0 +1,99 @@
+import { Tags } from '../Tags';
+import icon from './MangaUpJapan.webp';
+import { Chapter, DecoratableMangaScraper, Manga, Page, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchCSS } from '../platform/FetchProvider';
+
+type JSONChapter = {
+    id: number,
+    name: string,
+    subName: string
+}
+
+type JSONManga = {
+    titleName: string,
+    titleId: number,
+    chapters: JSONChapter[]
+}
+
+type JSONPage = {
+    content: {
+        value: {
+            imageUrl: string,
+        }
+    }
+}
+
+const mangasEndpoints = [
+    '/mon',
+    '/tue',
+    '/wed',
+    '/thu',
+    '/fri',
+    '/sat',
+    '/sun',
+    '/end',
+    '/yomikiri'
+].map(slug => `/series${slug}`);
+
+function MangasExtractor(element: HTMLAnchorElement) {
+    return {
+        id: element.pathname.match(new RegExpSafe('^/titles/(\\d+)$')).at(1),
+        title: element.querySelector('div.pc\\:text-title-md-pc').textContent.trim()
+    };
+}
+
+@Common.MangasSinglePagesCSS(mangasEndpoints, 'a:has(div.pc\\:text-title-md-pc)', MangasExtractor)
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+    public constructor() {
+        super('mangaupjapan', 'MangaUp (マンガアップ!)', 'https://www.manga-up.com', Tags.Language.Japanese, Tags.Source.Official, Tags.Media.Manga);
+    }
+
+    public override get Icon(): string {
+        return icon;
+    }
+
+    private ExtractNextJsPayloadData(element: HTMLScriptElement): string {
+        const data = element.innerHTML.match(/^self\.__next_f\.push\(\[\d+,"(.*)"\]\)$/)?.at(1);
+        return data?.replace(/\\{1,2}"/g, '"').replace(/\\{2,3}n/g, '\\n');
+    }
+
+    private async ExtractJSONData<T>(uri: URL, scriptMatcher: string, dataRegexp: RegExp): Promise<T> {
+        const request = new Request(uri.href);
```

Comment:
There is not much benefit of the intermediate variable `request`. May just inline the `new Request` into `FetchCSS`